src/Microsoft.AspNetCore.DataProtection/TypeForwardingActivator.cs

```diff
@@ -39,16 +40,14 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
                 forwardedTypeName = originalTypeName.Replace(OldNamespace, CurrentNamespace);
             }
 
-#if NET46
-            if (candidate || forwardedTypeName.Contains(CurrentNamespace))
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
             {
-                candidate = true;
-                forwardedTypeName = RemoveVersionFromAssemblyName(forwardedTypeName);
+                if (candidate || forwardedTypeName.Contains(CurrentNamespace))
```

Comment:
This check looks suspicious for two reasons:

1. Wrong culture used
2. To check the namespace, you want to check something more like `forwardedTypeName.StartsWith(CurrentNamespace + ".", StringComparison.Ordinal)`

... and please check all other places in DataProtection where maybe the wrong culture is being used.